File: first_scala/src/main/java/com/cpucode/student/Student.java
package com.cpucode.student;

/**
 * @author : cpucode
 * @date : 2021/11/21 15:56
 * @github : https://github.com/CPU-Code
 * @csdn : https://blog.csdn.net/qq_44226094
 */
public class Student {
    private String name;
    private Integer age;
    private static String school = "cpuCode";

    public Student(String name, Integer age){
        this.name = name;
        this.age = age;
    }

    public void printInfo() {
        System.out.println(this.name + " " + this.age + " " + Student.school);
    }

    public static void main(String[] args) {
        Student cpu = new Student("cpu", 22);
        Student code = new Student("code", 25);

        cpu.printInfo();
        code.printInfo();
    }
}

File: first_scala/src/main/scala/com/cpucode/student/Student.scala
package com.cpucode.student

/**
 * @author : cpucode
 * @date : 2021/11/21 16:00
 * @github : https://github.com/CPU-Code
 * @csdn : https://blog.csdn.net/qq_44226094
 */
class Student(name: String, age : Int) {
  def printInfo(): Unit = {
    println(name + " " + age + " " + Student.school)
  }
}

/**
 * 引入伴生对象
 */
object Student {
  val school: String = "cpuCode"

  def main(args: Array[String]): Unit = {
    val cpu = new Student("cpu", 22)
    val code = new Student("code", 25)

    cpu.printInfo()
    code.printInfo()
  }
}
